Exclude ROLE_DEFAULT from RoleMatrixType

Hanmac opened the pull request:
<!-- THE PR TEMPLATE IS NOT AN OPTION. DO NOT DELETE IT, MAKE SURE YOU READ AND EDIT IT! -->
## Subject

<!-- Describe your Pull Request content here -->

<!--
    Show us you choose the right branch.
    Different branches are used for different things :
    - 5.x is for everything backwards compatible, like patches, features and deprecation notices
    - 6.x is for deprecation removals and other changes that cannot be done without a BC-break
    More details here: https://github.com/sonata-project/SonataUserBundle/blob/5.x/CONTRIBUTING.md#base-branch
-->
I am targeting this branch, because it should be BC.

Per default only `UserInterface::ROLE_DEFAULT` aka `ROLE_USER` is excluded, but users can still use it.

If this Component would be moved outside of UserBundle that might be changed.

<!--
    Specify which issues will be fixed/closed.
    Remove it if this is not related.
-->

Closes #1535.

## Changelog

<!-- MANDATORY
    Fill the changelog part inside the code block.
    Follow this schema: http://keepachangelog.com/
    This will end up on https://github.com/sonata-project/SonataUserBundle/releases,
    please keep it short and clear and to the point
-->

<!--
    If you are updating something that doesn't require
    a release, you can delete the whole "Changelog" section.
    (eg. update to docs, tests)
-->

<!-- REMOVE EMPTY SECTIONS -->
```markdown
### Added
- Added `RoleMatrixType:exclude` Option to hide choices.

```

<!--
    If this is a work in progress, uncomment the "To do" section.
    You can add as many tasks as you want.
    If some are not relevant, just remove them.
-->
<!--
## To do

- [x] Update the tests;
- [ ] Update the documentation;
- [ ] Add an upgrade note.
-->

Hanmac commented:
@jordisala1991 is this good for a fix or should i extend the test cases more for the excluded option?

or if for example i set the excluded option to only `[Role_A]` should `UserInterface::ROLE_DEFAULT` still be excluded or not?

VincentLanglet commented on src/Form/Type/RolesMatrixType.php:
```diff
@@ -73,6 +75,7 @@ static function (Options $options, $value) {
 
                     return $value;
                 },
+            'exclude' => [UserInterface::ROLE_DEFAULT],
```
```suggestion
            'excluded_roles' => [UserInterface::ROLE_DEFAULT],
```
to be more specific ?

jordisala1991 replied:
Also please add docs about this feature

Hanmac replied:
what is your opinions about always excluding `UserInterface::ROLE_DEFAULT` in addition to `excluded_roles` ?

jordisala1991 replied:
I would prefer it to be configurable, so no auto addition, wdyt @VincentLanglet ?

VincentLanglet replied:
I agree, seems better to let it always configurable

Hanmac commented:
* added docs to the rst ones
* used better `excluded_roles` option name
* added more test cases

is that clear or should i add more config option? like defining it an Configuration option?

jordisala1991 commented:
For each option you add to an option resolver is a good practice to provide its allowed types, in this case it should be 'string[]'. Can you add that option?

Hanmac commented:
Your Opinion if that option should be configuable over the Config yml too?

VincentLanglet commented:
> Your Opinion if that option should be configuable over the Config yml too?

I don't think this is needed so far. Less config option is better if not needed.

VincentLanglet commented:
Is it ok for you @jordisala1991 ?
